### PR TITLE
fix(editor): Improve touch device detection

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
Only detecting touch devices that don't have fine precision input device (like mouse or trackpad).

## Related tickets and issues
Fixes ADO-1834

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

File: packages/editor-ui/src/composables/__tests__/useCanvasPanning.test.ts [hunk 1]
```diff
@@ -20,6 +20,11 @@ describe('useCanvasPanning()', () => {
 	let element: HTMLElement;
 	let elementRef: Ref<null | HTMLElement>;
 
+	beforeAll(() => {
+		// Mocks for useDeviceSupport
+		Object.defineProperty(window, 'matchMedia', { value: vi.fn().mockImplementation(() => true) });
```
Comment (alexgrozav) [inline packages/editor-ui/src/composables/__tests__/useCanvasPanning.test.ts:25]: Can we make this global in `src/__tests__/setup.ts`? Looks like it'll be needed whenever we use `useDeviceSupport`.

**Review by alexgrozav — CHANGES_REQUESTED** [review]
Looks good, one small remark.

File: packages/editor-ui/src/composables/__tests__/useCanvasPanning.test.ts (hunk 1 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/composables/__tests__/useCanvasPanning.test.ts:25]: Makes sense, will update

**Review by alexgrozav — APPROVED** [review]
Thanks for addressing all the feedback!

**Comment by janober** [comment]
Got released with [`n8n@1.47.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.47.0)

**Comment by janober** [comment]
Got released with [`n8n@1.47.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.47.0)

**Pull request merged**